File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Vpc/AlibabaCloud/SDK/ROS/CDK/Vpc/SslVpnClientCertProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Vpc
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::VPC::SslVpnClientCert`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-vpc.SslVpnClientCertProps")]
    public class SslVpnClientCertProps : AlibabaCloud.SDK.ROS.CDK.Vpc.ISslVpnClientCertProps
    {
        /// <summary>Property sslVpnServerId: ID of the SSL-VPN server.</summary>
        [JsiiProperty(name: "sslVpnServerId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOverride: true)]
        public object SslVpnServerId
        {
            get;
            set;
        }

        /// <summary>Property name: The name of the client certificate.</summary>
        /// <remarks>
        /// The length is 2-128 characters and must start with a letter or Chinese. It can contain numbers, periods (.), underscores (_), and dashes (-). But it can't start with http:// or https://.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "name", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true, isOverride: true)]
        public object? Name
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdkvpc/SslVpnClientCertProps.go
package alicloudroscdkvpc


// Properties for defining a `SslVpnClientCert`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-vpc-sslvpnclientcert
type SslVpnClientCertProps struct {
	// Property sslVpnServerId: ID of the SSL-VPN server.
	SslVpnServerId interface{} `field:"required" json:"sslVpnServerId" yaml:"sslVpnServerId"`
	// Property name: The name of the client certificate.
	//
	// The length is 2-128 characters and must start with a letter or Chinese. It can contain numbers, periods (.), underscores (_), and dashes (-). But it can't start with http:\/\/ or https:\/\/.
	Name interface{} `field:"optional" json:"name" yaml:"name"`
}

